var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -174,6 +173,20 @@ def determine_variants(cls, exes, version):
         return results
 
     def setup_build_environment(self, env):
+        if self.spec.satisfies("fabrics=cxi"):
+
+            def add_ldflags(name):
+                lib_dir = self.spec[name].libs.directories[0]
+                env.append_flags("LDFLAGS", f"-L{lib_dir}")
+                env.append_flags("LDFLAGS", f"-Wl,-rpath={lib_dir}")
+
+            env.append_flags("CFLAGS", f"-I{self.spec['json-c'].prefix.include}")
+            env.append_flags("CFLAGS", f"-I{self.spec['curl'].prefix.include}")
+            add_ldflags("json-c")
+            add_ldflags("curl")
+            if self.spec.satisfies("^curl +nghttp2"):
```

Comment:
This shouldn't be here. Looks like in @hppritcha's case curl was linked dynamically, and the linker tried to resolve nghttp symbols from transitive deps but couldn't. That's a curl problem, not a libfabrics issue.

Reply:
Also @hppritcha do you still see this issue where you cannot link to `libcurl.so`? If so can you share the spec?

Reply:
I ran into the same issue. `curl` wasn't external, but I need to retest now that we have proper configure flags passed to libfabric.

Reply:
At least on my end, the explicit configure options, removing the manual LDFLAGS and changing the dependency type has now resolved the issue. @hppritcha could you check it again with the current state?
@etiennemlb for me it looks like that `libfabric.so` now has the correct rpaths even without having to manually add them.

Reply:
i'll try to get to this tomorrow - 12/10/24